frontend/src/pages/Rider/Schedule.tsx

```diff
@@ -5,54 +5,121 @@ import React, {
+    const x = new Date(a.startTime);
+    const y = new Date(b.startTime);
+    if (x < y) return -1;
+    if (x > y) return 1;
+    return 0;
+  };
+
+  const datetime = new Date();
 
   const refreshRides = useCallback(() => {
-    fetch(`/api/rides?rider=${id}`, withDefaults())
+    fetch(`/api/rides?rider=${riderId}`, withDefaults())
       .then((res) => res.json())
-      .then(({ data }) => componentMounted.current && setRides([...data]));
+      .then(({ data }) => setRides([...data]));
   }, [id, withDefaults]);
 
   useEffect(() => {
     refreshRides();
+  }, [refreshRides, scheduledRides, unscheduledRides, setRides]);
+
+  console.log(rides);
+  const sorted = rides?.sort(compRides);
+  console.log('sorted');
+  console.log(sorted);
+  useEffect(() => {
+    const filteredRides = sorted?.filter((r) => {
+      const end = new Date(r.endTime);
+      return r.rider.email == email && end > datetime;
+    });
+    setFilteredRides(filteredRides);
 
-    return () => {
-      componentMounted.current = false;
-    };
-  }, [refreshRides]);
+    const filteredPastRides = sorted?.filter((r) => {
+      const end = new Date(r.endTime);
+      return r.rider.email == email && end < datetime;
+    });
+    setPastRides(filteredPastRides);
+
+    setAbsentRides(
+      sorted?.filter((r) => {
+        return r.status === Status.NO_SHOW;
+      })
+    );
+  }, [
+    refreshRides,
+    scheduledRides,
+    unscheduledRides,
+    rides,
+    setAbsentRides,
+    setFilteredRides,
+    setRides,
+  ]);
+  console.log('Filtered');
+  console.log(filteredRides);
+  filteredRides?.forEach((r) =>
+    console.log(new Date(r.endTime).toLocaleString())
+  );
 
   return (
     <main id="main">
-      <div className={styles.pageTitle}>
-        <h1 className={styles.header}>Hi {user?.firstName ?? ''}</h1>
-        <div className={styles.rightSection}>
-          <RequestRideModal onSubmit={refreshRides} />
-          <Notification />
-        </div>
-      </div>
-      {rides && rides.length > 0 && (
-        <>
-          <Collapsible title={'Your Upcoming Rides'}>
-            <RiderScheduleTable data={rides} isPast={false} />
-          </Collapsible>
-          <Collapsible title={'Your Past Rides'}>
-            <RiderScheduleTable data={rides} isPast={true} />
-          </Collapsible>
-        </>
+      {filteredRides && filteredRides.length >= 0 && (
+        <Collapsible title={'Your Upcoming Rides'}>
+          <RiderScheduleTable
+            data={filteredRides}
+            isPast={false}
+            email={email}
+          />
+        </Collapsible>
+      )}
+      {pastRides && pastRides.length >= 0 && (
+        <Collapsible title={'Your Past Rides'}>
+          <RiderScheduleTable data={pastRides} isPast={false} email={email} />
+        </Collapsible>
+      )}
+
+      <label>
+        <input
+          type="checkbox"
+          checked={filter}
+          onChange={() => setFilter(!filter)}
+          style={{
```

Comment:
This should be moved into a seperate CSS file

Reply:
second that! maybe you could create a CSS file if not already exists because random blocks of styling code in the main files is probably not good practices